### ISSUE TITLE
p2p_config host and port not working as expected

### ISSUE THREAD

**Issue by antiochp** [mimblewimble/grin#234]
grin.toml contains the following default config - 
```
[server.p2p_config]
host = "127.0.0.1"
port = 13414
```

The host and port combo are used for two things - 
1) to define the host and port that the p2p server will listen on _and_
2) we send this across in the handshake as the self_addr to other peers

So if we are running across a network it would seem reasonable to configure it like this - 

```
[server.p2p_config]
host = "0.0.0.0"
port = 13414
```

But this means other peers are not correctly seeing this peer as the `self_addr` is now incorrect (nobody else knows what 0.0.0.0 is here).

**Comment by ignopeverell** [comment]
I can take a look. The current code uses `is_loopback()` but I'm guessing `0.0.0.0` doesn't qualify. Should be a quick fix.

**Comment by antiochp** [comment]
I think we're going to need to explicitly define our own ip address (which public ip address other peers can find us on).
So we listen on `0.0.0.0`, but we can be found on, say `192.81.220.194`.

**Comment by antiochp** [comment]
Unless I'm misunderstanding something this is not something that will ever work without specifying the `addr` explicitly server by server.
There's no way to know which public IP address the server *should* be available on without specifying it in the config.

**Comment by antiochp** [comment]
Yeah I see now - so we just need to handle 0.0.0.0 and pull the ip from `peer_addr` like we do for 127.0.0.1

**Comment by ignopeverell** [comment]
Fixed in 15705b3
